hell/adventcalendar/2025/16/index.md

```diff
@@ -1,35 +1,114 @@
+* A dialog must be explicitly closed, either by taking an action within the dialog or closing via a close button, while a popover can be lightly dismissed by clicking outside of the popover.
+* A dialog can either be modal or non-modal, while a popover is always non-modal by design. As of late 2025 the Invoker Commands API only has a built-in command for a modal dialog, so I will be focusing on modal dialogs only in this post.
+* Dialogs have a built in backdrop functionality to tint the background a different color, while popovers shouldn't use a backdrop in most cases.
+* A dialog is an HTML element with a role of `dialog`, while a popover is considered an attribute and does not come with a role.
 
-Some text.
+## The `command` and `commandfor` attributes
+The Invoker Commands API introduces two new attributes to the `<button>` element—`command` and `commandfor`.  The `commandfor` attribute acts as a connector between the `<button>` controlling the functionality and the element that the command acts upon, while `command` contains the action that should be taken on the element.
 
+As of late 2025 the following commands are supported within the API, which right now can only be used for `<dialog>` or `<div role="dialog">` elements or on elements with the `popover` attribute. These commands are:
 
-<p class="highlight"><strong>Note:</strong> Some text.</p>
+* `show-modal`: Shows a `<dialog>` element as a modal. If you're familiar with the underlying API to show a `<dialog>` as a modal, this is equivalent to the `HTMLDialogElement.showModal()` method.
+* `close`: Closes a `<dialog>` element. This is equivalent to the `HTMLDialogElement.close()` method.
+* `request-close`: This command works similarly to the `close` command, but before the `<dialog>` can be closed, a `cancel` event is triggered first to cancel the closure of the dialog if required actions do not occur, such as selecting required options within the `<dialog>`. This is equivalent to the `HTMLDialogElement.requestClose()` method. This concept is a little tricky, so I'll show an example later in the post. 
+* `show-popover`: Shows a popover element. This is equivalent to both the `popoveraction="show"` attribute and the `HTMLElement.showPopover()` method.
+* `hide-popover`: Hides a popover element. This is equivalent to both the `popoveraction="hide"` attribute and the `HTMLElement.hidePopover()` method.
+* `toggle-popover`: Toggles a popover element; if the popover is hidden, it will be shown, and if the popover is shown, it will be hidden. This is equivalent to both the `popoveraction="toggle"` attribute and the `HTMLElement.togglePopover()` method.
+* Custom values: This is where The Invoker API really shines! A user can provide a custom value, prefixed by two hyphens (`--`) to create a custom `command` event. This event is usually written in JavaScript and provided by the user to add new functionality. The possibilities are endless!
```

Comment:
later in the post you mention some examples, could add a 'more on that later' forward reference here?